keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Resource.kt

```diff
@@ -30,6 +30,7 @@ data class Resource<T : Any>(
     require(kind.isNotEmpty()) { "resource kind must be defined" }
     require(metadata["uid"].isValidULID()) { "resource uid must be a valid ULID" }
     require(metadata["name"].isValidName()) { "resource name must be a valid name" }
+    require(metadata["serviceAccount"].isValidServiceAccount()) { "serviceAccount must be a valid email" }
```

Comment:
Fiat doesn't require service account names to be in the form of an email address. We have at least one that isn't internally. I think this can just be dropped. Since ```@PreAuthorize("@authorizationSupport.userCanModifySpec(#submittedResource.metadata.serviceAccount)")``` is on the create controller, that should guarantee that  `serviceAccount` is valid according to fiat regardless of formatting.

Reply:
Oo good point. I'll update.